### PR TITLE
Add support ZFS filesystem

### PR DESCRIPTION
**Pull request by miguel250**
To access ZFS stats from a container, it needs to have the `privileged` flag and `zfsutils`needs to be installed in the container. @vishh 

```bash
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 1196BA81F6B0FC61
sudo apt-get install python-software-properties
sudo apt-add-repository ppa:zfs-native/stable
apt-get update
apt-get install zfsutils

```

### CODE REVIEW

File: fs/fs.go [hunk 1]
```diff
@@ -128,7 +135,7 @@ func getDockerImagePaths(context Context) []string {
 	// TODO(rjnagal): Detect docker root and graphdriver directories from docker info.
 	dockerRoot := context.DockerRoot
 	dockerImagePaths := []string{}
-	for _, dir := range []string{"devicemapper", "btrfs", "aufs", "overlay"} {
+	for _, dir := range []string{"devicemapper", "btrfs", "aufs", "overlay", "zfs"} {
```
Comment (vishh) [inline fs/fs.go]: Is ZFS a storage driver in docker?
I thought we can use ZFS instead of ext4 as the base fs.

Comment (vishh) [inline fs/fs.go]: nvm. Just noticed this page: https://docs.docker.com/engine/userguide/storagedriver/zfs-driver/

File: container/docker/fsHandler.go [hunk 2]
```diff
@@ -96,6 +96,7 @@ func (fh *realFsHandler) trackUsage() {
 }
 
 func (fh *realFsHandler) start() {
+	fh.update()
```
Comment (vishh) [inline container/docker/fsHandler.go]: nit: Can we move this to `fh.trackUsage`?

**Comment by vishh** [comment]
What about handling capacity and usage for the base filesystem under zfs? Is it already working ?

**Comment by miguel250** [comment]
@vishh Yeah this set as well.

**Comment by vishh** [comment]
We should add an e2e test for this. Can you file an issue?

**Comment by vishh** [comment]
ok to test

**Comment by miguel250** [comment]
Just created #1021

**Comment by vishh** [comment]
ok to test

**Comment by vishh** [comment]
@jimmidyson: I wonder why the network tests are flaky :(

The e2e failures are unrelated to this PR. So this PR LGTM

**Comment by miguel250** [comment]
Hi @vishh,
is there anything else that needs to be done for this pull request?

**Comment by vishh** [comment]
Oops. This fell out of my radar. Sorry!

**Comment by vishh** [comment]
@cadvisorJenkinsBot: test this please

**Comment by vishh** [comment]
ok to test

**Comment by DanielDent** [comment]
I'm unclear if zfs support is supposed to be working right now. For me, it does not appear to be.. On systems with ZFS, I get complaints about /dev/zfs being missing. If I provide the /dev/zfs device by bind mounting the host's device, then I get new complaints about zfs libraries being missing.

**Pull request merged**